nixos/modules/services/computing/boinc/client.nix

```diff
@@ -6,6 +6,24 @@ let
   cfg = config.services.boinc;
   allowRemoteGuiRpcFlag = optionalString cfg.allowRemoteGuiRpc "--allow_remote_gui_rpc";
 
+  # A package providing the boinc_client that will actually be executed.
+  execPkg = if cfg.useFHSEnv then
+    pkgs.buildFHSUserEnv {
+      name = "boinc_client";
+      targetPkgs = p: [ cfg.package ] ++ optional cfg.virtualbox.enable cfg.virtualbox.package;
+      multiPkgs = p: optional cfg.gpu.nvidia.enable cfg.gpu.nvidia.package;
+      runScript = "/bin/boinc";
+      extraBuildCommands = ''
+        if [ -e lib/vendors ]; then
+          chmod +w etc
+          mkdir -p etc/OpenCL
+          ln -s $(realpath lib/vendors) etc/OpenCL/vendors
+          chmod -w etc
+        fi
+      '';
```

Comment:
`lib/vendors` comes from the `nvidia-x11` package (and probably also the packages that provide the OpenCL implementations for AMD and Intel hardware too). It contains ICD files, which contain some information about how the generic OpenCL API is mapped to the device driver. Off the top of my head, I believe `nvidia-x11` provides one ICD file, `nvidia.icd`, which is necessary for the OpenCL API to detect NVIDIA GPUs.

`lib/vendors` is just where this ICD file is currently placed by the derivation of `nvidia-x11`. When I made this change I went for the less destructive option, but in my opinion `etc/OpenCL/vendors` would be the correct location for this file.